assets/src/edit-story/components/checklist/index.js

```diff
@@ -13,4 +13,94 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-//@TODO export ppc popup component here from behind a feature flag
+/**
+ * External dependencies
+ */
+import { __ } from '@web-stories-wp/i18n';
+import { useState, useEffect, useRef } from 'react';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useFocusOut } from '../../../design-system';
+import { FOCUSABLE_SELECTORS } from '../../constants';
+import { useHelpCenter } from '../../app';
+import DirectionAware from '../directionAware';
+import { Popup } from '../helpCenter/popup';
+import { NavigationWrapper } from '../helpCenter/navigator';
+import { TopNavigation } from '../helpCenter/navigator/topNavigation';
+import { Z_INDEX } from '../canvas/layout';
+import { Toggle } from './toggle';
+import { POPUP_ID } from './constants';
+import { DesignChecks } from './designChecks';
+import { AccessibilityChecks } from './accessibilityChecks';
+import { PriorityChecks } from './priorityChecks';
+
+const Wrapper = styled.div`
+  /**
+   * sibling inherits parent z-index of Z_INDEX.EDIT
+   * so this needs to be placed above that while still
+   * retaining its position in the DOM for focus purposes
+   */
+  z-index: ${Z_INDEX.EDIT + 1};
+`;
+
+export function Checklist() {
+  const navRef = useRef();
+  const wrapperRef = useRef();
+  const [isOpen, setIsOpen] = useState(false);
+  const { close: closeHelpCenter } = useHelpCenter(
+    ({ actions: { close }, state: { isOpen: isHelpCenterOpen } }) => ({
+      close,
+      isHelpCenterOpen,
+    })
+  );
+
+  // Set Focus within the popup on open
+  useEffect(() => {
+    if (isOpen) {
+      const firstFocusableChild = navRef.current?.querySelector(
+        FOCUSABLE_SELECTORS.join(', ')
+      );
+      firstFocusableChild?.focus();
+    }
+  }, [isOpen]);
+
+  // Close this popup when focus leaves
+  // this also functions to close this popup
+  // when either the help center or keyboard
+  // actions are open.
+  useFocusOut(wrapperRef, () => setIsOpen(false), []);
+
+  return (
+    <DirectionAware>
+      <Wrapper ref={wrapperRef}>
+        <Popup
+          popupId={POPUP_ID}
+          isOpen={isOpen}
+          ariaLabel={__('Checklist', 'web-stories')}
+        >
+          <NavigationWrapper ref={navRef}>
+            <TopNavigation
+              onClose={() => setIsOpen(false)}
```

Comment:
💭 nit: If we're worried about extra re-renders, this could be wrapped in a `useCallback` since it'll be re-created every render.

Side note: interested in your thoughts about memoizing functions in every component. Maybe it's not necessary?

Reply:
Yea I felt like it wasn't a big deal here because the `TopNavigation` only has like 2 children. Personally I think it's necessary and good if it's higher up a large component tree, but if it's just got a couple children I think it's basically negligible with or without memoizing any props